### DISCUSSION TITLE
runcmd is not present on handlers list

### DISCUSSION THREAD

**Discussion by chaq3l** [naemon/naemon-core#445] [Q&A]
Currently when I run: ./naemon_core/contrib/naemon-qh.rb I get such response:

``
Welcome to Naemon Query handler, here's a list of handlers available:

merlin     Merlin information
wproc      Worker process management and info
nerd       Naemon Event Radio Dispatcher - Subscriber Service
help       Help for the query handler
echo       The Echo Service - What You Put Is What You Get
command    Naemon external commands interface

use @<handler> <cmd> for running queries such as '@nerd subscribe servicechecks'

quit or exit will quit interactive mode
``

Is there any way to activate or turn on runcmd to make it present on handlers list?

**Comment by nook24** [comment]
Hi, what does the `runcmd ` do?

**Reply by chaq3l** [reply]
https://github.com/naemon/naemon-core/blob/master/lib/runcmd.c
https://github.com/NagiosEnterprises/nagioscore/blob/master/lib/runcmd.c

Currently, the only information I have is the one in the code comments

**Reply by chaq3l** [reply]
I am trying to run "/opt/plugins/suid/check_host -H localhost" command from monitor ninja via this handler. I wonder if this handler was available in earlier versions of naemon. I don't want to re-implement something that would already be available in a different version of the code.

**Reply by sni** [reply]
i don't think it is necessary to use a naemon worker for this, you could simply run the command as is for testing or preview purposes. At least that's what Thruk does.

**Reply by chaq3l** [reply]
I found rpm that I am interested in:
[monitor-testthis.tar.gz](https://github.com/naemon/naemon-core/files/14095762/monitor-testthis.tar.gz)
After installation handler that I need is present :)

**Comment by sni** [comment]
What are you trying to solve? For running plugins, you don't have to fiddle with the query handler. As far as i remember, the `runcmd` thing is used internally by the naemon worker to actually execute the checks.

**Reply by chaq3l** [reply]
I am trying to reproduce the `test this check` functionality that I have seen in commercial solutions.
It calls e.g.: `/usr/lib64/nagios/plugins/check_ping -H localhost -w 3000.0,80% -c 5000.0,100% -p 5` using the runcmd handler.

**Reply by nook24** [reply]
I searched the code real quick but did not find any handler that would register a `runcmd` command or anything like that.

> I am trying to reproduce the test this check functionality that I have seen in commercial solutions.

I would implement the good old [SCHEDULE_FORCED_HOST_CHECK](https://www.naemon.io/documentation/developer/externalcommands/schedule_forced_host_check.html) external command to force the check execution.

Otherwise, if you want this for debugging of commands without the need to open up a ssh connection, you could simply uses `exec` and execute the command as `naemon` user by yourself. Probably much simpler and less hassle than fiddling around with the query handler.
